File: alimentos.py
import Productos.Productos

class Alimentos(Productos):
    def __init__(self, nombre, precio, fecha_de_importacion, stock, fecha_de_caducidad):
        super().__init__(nombre, precio, fecha_de_importacion, stock)
        self.fecha_de_caducidad = fecha_de_caducidad
        self.productos_alimentos = []

    def registrar_producto_alimento(self, nombre, precio, fecha_de_importacion, stock):
        print("Ingrese la fecha de caducidad con la siguiente estructura: día/mes/año, ejemplo: 06/10/24")
        self.fecha_de_caducidad = input()
        nuevo_alimento = Alimentos(nombre, precio, fecha_de_importacion, stock, self.fecha_de_caducidad)
        self.productos_alimentos.append(nuevo_alimento)

    def get_productos_alimentos(self):
        return self.productos_alimentos

File: src/Productos/alimentos.java
package Productos;

import java.util.ArrayList;
import java.util.Scanner;
public class alimentos extends Productos{
    Scanner leer=new Scanner(System.in);
    ArrayList <alimentos> ProductosAlimentos=new ArrayList<>();
    private String FechaDeCaducidad;
    public alimentos(String nombre, double precio, String fechaDeImportacion, int stock, String fechaDeCaducidad) {
        super(nombre, precio, fechaDeImportacion, stock);
        this.FechaDeCaducidad = fechaDeCaducidad;
    }
    public void RegistrarProductoAlimento(String nombre,double precio, String fechaDeImportacion, int stock){
        System.out.println("ingrese la fecha de caducidad con la siguiente estructura: dia/mes/año, ejepmlo: 06/10/24");
        FechaDeCaducidad=leer.nextLine();
        alimentos nuevoAlimento=new alimentos(nombre, precio,fechaDeImportacion, stock, FechaDeCaducidad);
        ProductosAlimentos.add(nuevoAlimento);
    }
    public ArrayList<alimentos> getProductosAlimentos() {
        return ProductosAlimentos;
    }

}
